lib/ecto/pagination.ex

```diff
@@ -0,0 +1,79 @@
+defmodule PhoenixApiToolkit.Ecto.Pagination do
+  @moduledoc """
+  Helper functions for pagination.
+  """
+  import Ecto.Query
+
+  @doc """
+  Paginate the query by specifying a limit and/or offset.
+
+  Note: must be called after all filters have been added to the query.
+
+  The limit and/or offset will be applied to the primary model.
+
+      iex> base_query()
+      ...> |> paginate(%{})
+      #Ecto.Query<from u0 in "users", as: :user>
+
+      iex> base_query()
+      ...> |> paginate(%{limit: 10})
+      #Ecto.Query<from u0 in "users", as: :user, where: u0.id in subquery(from u0 in "users",
+        as: :user,
+        limit: ^10,
+        select: u0.id)>
+
+      iex> base_query()
+      ...> |> paginate(%{offset: 10})
+      #Ecto.Query<from u0 in "users", as: :user, where: u0.id in subquery(from u0 in "users",
+        as: :user,
+        offset: ^10,
+        select: u0.id)>
+
+      iex> base_query()
+      ...> |> paginate(%{limit: 10, offset: 10})
+      #Ecto.Query<from u0 in "users", as: :user, where: u0.id in subquery(from u0 in "users",
+        as: :user,
+        limit: ^10,
+        offset: ^10,
+        select: u0.id)>
+
+      iex> query = base_query()
+      ...> |> join(:left, [u], e in assoc(u, :email_addresses), as: :email_addresses)
+      ...> |> where([email_addresses: e], e.is_verified)
+      ...> |> paginate(%{offset: 20, limit: 10})
+      #Ecto.Query<from u0 in "users", as: :user, left_join: e1 in assoc(u0, :email_addresses), as: :email_addresses, where: e1.is_verified, where: u0.id in subquery(from u0 in "users",
+        as: :user,
+        left_join: e1 in assoc(u0, :email_addresses),
+        as: :email_addresses,
+        where: e1.is_verified,
+        limit: ^10,
+        offset: ^20,
+        select: u0.id)>
+  """
+  @spec paginate(Ecto.Query.t(), map) :: Ecto.Query.t()
+  def paginate(query, filters) do
```

Comment:
ik zou graag de signature willen veranderen in (query, filters, named_binding) zodat je kunt specificeren op welke binding er pagination plaats moet vinden. Dit maakt het explicieter.